<launch>
  <param name="/use_sim_time" value="true" />

  <node name="pso_slam_node" pkg="pso_slam" type="pso_slam_" output="screen">
    <rosparam param="scan_topic">"/laser_1/scan"</rosparam>
    <rosparam param="scan_frame_id">"laser_1"</rosparam>
    <rosparam param="base_frame_id">"base_link"</rosparam>
    <rosparam param="global_frame_id">"map"</rosparam>
    <rosparam param="odom_frame_id">"odom"</rosparam>
    <rosparam param="map_size_m">300</rosparam>
    <rosparam param="pso_num_threads">4</rosparam>
    <rosparam param="pso_iterations">30</rosparam>
    <rosparam param="pso_population">50</rosparam>
    <rosparam param="frame_size">300</rosparam>
    <rosparam param="cell_size">0.5</rosparam>
    <rosparam param="rate_hz">10</rosparam>
    <rosparam param="initial_pose_x">0.0</rosparam>
    <rosparam param="initial_pose_y">0.0</rosparam>
    <rosparam param="initial_pose_a">0.0</rosparam>
  </node>

  <!-- Publish the associated static transform -->
  <node name="static_tf" pkg="tf" type="static_transform_publisher" output="screen" args="0.65 0 0 0 0 0 base_link laser_1_mount_link 100" />
</launch>
